pkg/resources/kafka/kafka.go

```diff
@@ -655,7 +652,7 @@ func (r *Reconciler) reconcileKafkaPod(log logr.Logger, desiredPod *corev1.Pod)
 		//Since toleration does not support patchStrategy:"merge,retainKeys", we need to add all toleration from the current pod if the toleration is set in the CR
 		if len(desiredPod.Spec.Tolerations) > 0 {
 			desiredPod.Spec.Tolerations = append(desiredPod.Spec.Tolerations, currentPod.Spec.Tolerations...)
-			uniqueTolerations := []corev1.Toleration{}
+			uniqueTolerations := make([]corev1.Toleration, len(desiredPod.Spec.Tolerations), 0)
```

Comment:
```suggestion
			uniqueTolerations := make([]corev1.Toleration, 0, len(desiredPod.Spec.Tolerations))
```

Reply:
Thanks nice catch this is a serious error.